intro-HPC/examples/SCOOP/picalc.py

```diff
@@ -0,0 +1,38 @@
+from math import hypot
+from random import random
+from scoop import futures
+
+NAME = 'SCOOP_piCalc'
+
+# A range is used in this function for python3. If you are using python2,
+# an xrange might be more efficient.
+try:
+    range_fn = xrange
+except:
+    range_fn = range
+
+
+def test(tries):
+    return sum(hypot(random(), random()) < 1 for i in range_fn(tries))
+
+# Calculates pi with a Monte-Carlo method. This function calls the function
+# test "n" times with an argument of "t". Scoop dispatches these
+# functions interactively accross the available resources.
+def calcPi(workers, tries):
+    expr = futures.map(test, [tries] * workers)
+    piValue = 4. * sum(expr) / float(workers * tries)
+    return piValue
+
+
+if __name__ == '__main__':
+    import sys
+    nr_batches = 3000
+    batch_size = 5000
+
+    try:
+        nr_batches = int(sys.argv[1])
+        batch_size = int(sys.argv[2])
+    except:
+        pass
```

Comment:
@redfast00 Please change this to spit out a meaningful error if the program is fed wrong arguments? You can use `if sys.argv[1:]:` to only try to parse the argument if any are provided.